src/sf/grid/social-media/posts.twitter.suma

```diff
@@ -0,0 +1,115 @@
+// Twitter API Reference: https://developer.twitter.com/en/docs/twitter-api
+
+profile = "social-media/posts@1.0"
+provider = "twitter"
+
+map GetProfilePosts {
+
+  http GET "/2/users/{input.profileId}/tweets" {
+
+    request {
+      query {
+        'tweet.fields' = 'id,text,created_at,attachments'
+        expansions = 'attachments.media_keys'
+        'media.fields' = 'duration_ms,height,media_key,preview_image_url,type,url,width,alt_text'
+        start_time = input.afterDate
+        end_time = input.beforeDate
+        pagination_token = input.page
+        max_results = 25
+      }
+      headers {
+        authorization = "Bearer " + parameters.accessToken
+      }
+    }
+
+    response 200 "application/json" {
+      posts = []
+
+      set if (body.meta.result_count > 0) {
+        //workaround map interpreter variables merging issue
+        posts = call foreach(index of Array(body.data.length - 1).fill().map((_, i) => i)) MapTweetToPost(
+            index = index,
+            tweets = body.data,
+            includes = body.includes
+          )
+      }
+
+      map result {
+        posts = posts
+        previousPage = body.meta.previous_token
+        nextPage = body.meta.next_token
+      }
+    }
+
+    response 400 "application/json" {
+      map error {
+        title = "Bad request"
+        detail = body.detail
+      }
+    }
+
+    response 401 "application/json" {
+      map error {
+        title = "Unauthenticated"
+        detail = body.detail
+      }
+    }
+
+    response 403 "application/json" {
+      map error {
+        title = "Forbidden"
+        detail = body.detail
+      }
+    }
+  }
+}
+
+operation MapTweetToPost {
+  index = args.index
+  tweet = args.tweets[index]
+  includes = args.includes
+  attachments = []
+
+   set if(tweet && tweet.attachments && tweet.attachments.media_keys && includes.media) {
+    attachments = call foreach(mediaKey of tweet.attachments.media_keys) MapMediaAttachment(
+        mediaKey = mediaKey,
+        media = includes.media
+      )
+  }
+
+  return {
+    id = tweet.id
+    url = 'https://twitter.com/twitter/status/' + tweet.id
```

Comment:
Oh, I see they even recommend this trick... https://blog.twitter.com/developer/en_us/topics/tips/2020/getting-to-the-canonical-url-for-a-tweet

I saw usually `i` being used for permalinking, i.e.:
```suggestion
    url = 'https://twitter.com/i/status/' + tweet.id
```